auth/service/user.go

```diff
@@ -13,26 +13,17 @@ import (
 	"time"
 
 	"github.com/allegro/bigcache/v3"
+	"github.com/jackc/pgx/v5"
 )
 
 func (d *AuthService) CreateUser(ctx context.Context, user *auth.User) (id int64, err error) {
-	conn, err := d.db.Conn(ctx)
-	if err != nil {
-		return 0, fmt.Errorf("acquiring connection pool: %w", err)
-	}
-	defer func() {
-		err := conn.Close()
-		if err != nil {
-			log.Printf("error closing connection: %v", err)
-		}
-	}()
-
-	tx, err := conn.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSnapshot})
+	tx, err := d.pool.Begin(ctx)
```

Comment:
beginTx, kasih IsoLevel. pake ReadCommitted, tambahin `AccessMode: pgx.ReadOnly`
```suggestion
	tx, err := d.pool.BeginTx(ctx, pgx.TxOptions{IsoLevel: pgx.ReadCommitted, AccessMode: pgx.ReadWrite})
```

Reply:
`ReadWrite` maksudnya?

Reply:
iya